test/functional/tests/data_integrity/test_data_integrity_with_patterns.py

```diff
@@ -0,0 +1,254 @@
+#
+# Copyright(c) 2020 Intel Corporation
+# SPDX-License-Identifier: BSD-3-Clause-Clear
+#
+
+import pytest
+from datetime import timedelta
+
+from api.cas import casadm
+from api.cas.cache_config import CacheMode
+from core.test_run import TestRun
+from test_tools.disk_utils import Filesystem
+from test_tools.fio.fio import Fio
+from test_tools.fio.fio_param import ReadWrite, IoEngine, CpusAllowedPolicy, ErrorFilter
+from test_tools.fio.fio_patterns import Pattern
+from storage_devices.disk import DiskType, DiskTypeSet, DiskTypeLowerThan
+from test_tools.fs_utils import remove
+from test_utils.size import Unit, Size
+
+exp_obj_size = Size(50, Unit.GibiByte)
+exp_obj_number = 4
+unique_patterns_number = len(Pattern.get_unique_patterns())
+runtime = timedelta(hours=2)
+mount_point = '/mnt/cas'
+
+
+@pytest.mark.parametrize(
+    "pattern", [Pattern.cyclic, Pattern.sequential]
+)
+@pytest.mark.parametrize("cache_mode", CacheMode)
+@pytest.mark.require_disk("cache", DiskTypeSet([DiskType.optane, DiskType.nand]))
+@pytest.mark.require_disk("core", DiskTypeLowerThan("cache"))
+def test_data_integrity_single_pattern(cache_mode, pattern):
+    """
+        title:
+          Data integrity test with different write patterns with 2 hours duration time
+        description: |
+          Run fio with single pattern on each core and verify if there are no write errors.
+        pass_criteria:
+            - System does not crash.
+            - Fio does not return any errors.
+    """
+    with TestRun.step("Prepare cache and core."):
+        cache_dev = TestRun.disks['cache']
+        cache_dev.create_partitions([exp_obj_size * (exp_obj_number * 1 / 4)])
+        cache_part = cache_dev.partitions[0]
+        core_dev = TestRun.disks['core']
+        core_dev.create_partitions([exp_obj_size] * exp_obj_number)
+
+    with TestRun.step(f"Start cache in {cache_mode} mode"):
+        cache = casadm.start_cache(cache_part, cache_mode, force=True)
+
+    with TestRun.step(f"Add all core devices to cache."):
+        cores = []
+        for core_part in core_dev.partitions:
+            core_part.create_filesystem(Filesystem.ext3, True)
+            cores.append(cache.add_core(core_part))
+
+    with TestRun.step("Mount all cores"):
+        fio_targets = []
+        for i, core in enumerate(cores):
+            core_mount_point = f"{mount_point}{cache.cache_id}-{core.core_id}"
+            core.mount(f"{core_mount_point}")
+            fio_targets.append(f"{core_mount_point}/test_file")
+
+    with TestRun.step(f"Run fio with {pattern.name} pattern."):
+        fio = prepare_unified_fio(fio_targets, pattern)
+        fio.run()
+
+    with TestRun.step("Check for fio errors."):
+        fio_output = TestRun.executor.run(f"cat {fio.fio.fio_file}")
+        fio_errors = fio.get_results(fio_output.stdout)[0].total_errors()
```

Comment:
`fio.run()` already returns `FioResult`, so you don't need to pull in this file and instantiate `FioResult` manually